File: banda/SergioDeIscar-banda-java/SergioDeIscar-banda-java/src/main/java/org/example/models/Guitarrista.java
package org.example.models;

import org.example.enums.GuitarraType;
import org.example.interfaces.IGuitarrista;

public final class Guitarrista extends Musician implements IGuitarrista {
    private final float salario = super.salario * 1.5f;
    private GuitarraType guitarra;
    public Guitarrista(String nombre, int experiencia, GuitarraType guitarra) {
        super(nombre, experiencia);
        setGuitarra(guitarra);
    }

    @Override
    public float getSalario() {
        return salario;
    }

    @Override
    public GuitarraType getGuitarra() {
        return guitarra;
    }

    @Override
    public void setGuitarra(GuitarraType guitarra) {
        this.guitarra = guitarra;
    }

    @Override
    public void guitarrear() {
        System.out.println("El guitarrista " + nombre + " esta guitarreando");
    }
}

File: banda/SergioDeIscar-banda-kotlin/SergioDeIscar-banda-kotlin/src/main/kotlin/models/Guitarrista.kt
package bandamusical.models

import bandamusical.interfeces.IGuitarrista

class Guitarrista(nombre:String,
                  override val experiencia: Int,
                  override val guitarra: GuitarraType): Musician(nombre), IGuitarrista {
    override val salario: Float = super.salario * 1.35f
    override fun guitarrear() {
        println("El guitarrista $nombre gitarrea, el tipo de gitarra utilizado es $guitarra.")
    }
}

enum class GuitarraType{
    CLASICA, //CLÁSICA
    ACUSTICA, //ACÚSTICA
    FLAMENCA,
    ELECTRICA, //ELÉCTRICA
    ELECTROACUSTICA, //ELECTROACÚSTICA
    SEMIACUSTICA, //SEMIACÚSTICA
    MIDI,
    ITALIANA
}